cmd/interceptors/main.go

```diff
@@ -82,13 +123,64 @@ func main() {
 		IdleTimeout:  idleTimeout,
 		Handler:      mux,
 	}
-
-	logger.Infof("Listen and serve on port %d", Port)
-	if err := srv.ListenAndServe(); err != nil {
+	logger.Infof("Listen and serve on port %d", HTTPSPort)
+	if err := srv.ListenAndServeTLS(certFile, keyFile); err != nil {
 		logger.Fatalf("failed to start interceptors service: %v", err)
 	}
+
 }
 
 func handler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
+
+// getCerts uses Knative pkg to generate certs for clusterinterceptor to run as https
+func getCerts(ctx context.Context, secretLister v1.SecretLister, kubeClient *kubeclientset.Clientset) (string, string, []byte, error) {
+	interceptorSvcName := os.Getenv("INTERCEPTOR_TLS_SVC_NAME")
+	interceptorSecretName := os.Getenv("INTERCEPTOR_TLS_SECRET_NAME")
+	namespace := os.Getenv("SYSTEM_NAMESPACE")
+
+	secret, err := secretLister.Secrets(namespace).Get(interceptorSecretName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// The secret should be created explicitly by a higher-level system
+			// that's responsible for install/updates.  We simply populate the
+			// secret information.
+			log.Printf("secret %s is missing", interceptorSecretName)
+			return "", "", []byte{}, err
+		}
+		log.Printf("error accessing certificate secret %q: %v", interceptorSecretName, err)
+		return "", "", []byte{}, err
+	}
+
+	serverKey, serverCert, caCert, err := certresources.CreateCerts(ctx, interceptorSvcName, namespace, time.Now().Add(decade))
```

Comment:
Question: Wouldn't this create a new cert on each restart - Also if we have multiple replicas of the Core interceptor server running, wouldn't they overwrite each other? I think we might want to see if the certs already exist and are valid, if they do then we should just use them instead of creating a new one each time

Reply:
Yes you are right

As discussed in Wg call added condition check to verify existence of certs and if it is there no need to create

**Pending:** Certs validation and Rotation will create an issue to track this

Thank you :+1: